lib/dock.rb

```diff
@@ -0,0 +1,34 @@
+class Dock
+  attr_reader :name, :max_rental_time, :rental_time
+  def initialize(name, max_rental_time)
+    @name = name
+    @max_rental_time = max_rental_time
+    @revenue_multiplier = []
+    @rental_time
+  end
+
+  def rent(boat, renter)
+    @revenue_multiplier << boat.price_per_hour
+    @rental_time = 0
+  end
+
+  def log_hour
+    @rental_time += 1
+  end
+
+  def return(boat)
```

Comment:
I'm not following your logic here - Ideally, when a boat is returned, it would remove that boat from some sort of rental collections.  Or, in your design, when a boat is returned, you would want to remove that boat's revenue multiplier from the `@revenue_multiplier` collection.